### PR TITLE
Husky config

### PR DESCRIPTION
**Pull request by marshall2112**
Dapp level config is moved to the monorepo root, as per https://github.com/lint-staged/lint-staged#how-to-use-lint-staged-in-a-multi-package-monorepo and https://typicode.github.io/husky/guide.html

### CODE REVIEW

**Comment by marshall2112** [comment]
> @marshall2112 would you add the [no-array-index-key eslint rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) to this PR (#904) or its a separate PR afterwards as it likely requires changes in code that are not purely aesthetic?

I was thinking to do that in a separate PR, yeah.

**Comment by nichosystem** [comment]
Deploy preview for _core_ ready!

✅ Preview
https://core-5iym3pmbt-templedao.vercel.app
https://pr-944-preview.templedao.link

Built with commit 10ef71fde2ba83ec3a356a8faa38e71d637e4fca.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

**Comment by marshall2112** [comment]
> @marshall2112 now that tlc and the grand refactor is live we might want to revert [the](https://github.com/TempleDAO/temple/pull/944/commits/11cc78049b18381d03e5f3a412c68439eaee8e35) rebase onto stage and re run the full format?

I went ahead and re-applied as you suggested.. Ready for review

**Comment by marshall2112** [comment]
> apps/dapp/abi
> .yarn
> protocol/typechain
> protocol/artifacts-hardhat

I understood that we were only applying prettier to the dapp project at first, then (optionally) expanding its use later. Let me take a look at applying it to the whole monorepo in one go.

File: apps/dapp/.lintstagedrc.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "*.{js,ts,tsx}": ["prettier --write", "eslint --fix"],
```
[the remark this answers is not in the document]
Comment (marshall2112) [reply apps/dapp/.lintstagedrc.json:2]: I've been trying to understand why I chose this order (a couple months ago), and to be honest, I can't. 

I think your logic makes sense, and I'll switch the order. 👍

**Comment by marshall2112** [comment]
> > I understood that we were only applying prettier to the dapp project at first, then (optionally) expanding its use later. Let me take a look at applying it to the whole monorepo in one go. 👍
> 
> ah np then! I agree scoping it to dapp only makes sense. Old contract deployment scripts/tests wont likely be touched and reformatted.

Oh I just pushed a commit to apply to the entire monorepo.

We can roll this back if preferred. Wdyt?

**Pull request merged**